libtransact/src/workload/batch_gen.rs

```diff
@@ -63,7 +63,12 @@ impl<'a> SignedBatchProducer<'a> {
         loop {
             match self.next() {
                 Some(Ok(batch)) => {
-                    if let Err(err) = batch.write_length_delimited_to_writer(writer) {
+                    let proto_batch = Batch::from_native(batch).map_err(|_| {
+                        BatchingError::InvalidStateError(InvalidStateError::with_message(
+                            "Failed to build batch".into(),
```

Comment:
This should probably be an InternalError constructed with InternalError::from_source(e) where e is the error being mapped.

Reply:
Fixed